pkg/verification/verification_service_test.go

```diff
@@ -0,0 +1,148 @@
+package verification_test
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/gin-gonic/gin"
+
+	"gopkg.in/h2non/gock.v1"
+
+	"github.com/codeready-toolchain/registration-service/pkg/configuration"
+
+	"github.com/codeready-toolchain/api/pkg/apis/toolchain/v1alpha1"
+	"github.com/stretchr/testify/require"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/codeready-toolchain/registration-service/pkg/verification"
+
+	"github.com/codeready-toolchain/registration-service/test"
+	"github.com/stretchr/testify/suite"
+)
+
+type TestVerificationServiceSuite struct {
+	test.UnitTestSuite
+}
+
+func TestRunVerificationServiceSuite(t *testing.T) {
+	suite.Run(t, &TestVerificationServiceSuite{test.UnitTestSuite{}})
+}
+
+type mockVerificationConfig struct {
+	accountSID      string
+	authToken       string
+	fromNumber      string
+	messageTemplate string
+}
+
+func (c *mockVerificationConfig) GetTwilioAccountSID() string {
+	return c.accountSID
+}
+
+func (c *mockVerificationConfig) GetTwilioAuthToken() string {
+	return c.authToken
+}
+
+func (c *mockVerificationConfig) GetTwilioFromNumber() string {
+	return c.fromNumber
+}
+
+func (c *mockVerificationConfig) GetVerificationMessageTemplate() string {
+	return c.messageTemplate
+}
+
+func NewMockVerificationConfig(accountSID, authToken, fromNumber string) verification.ServiceConfiguration {
+	return &mockVerificationConfig{
+		accountSID:      accountSID,
+		authToken:       authToken,
+		fromNumber:      fromNumber,
+		messageTemplate: configuration.DefaultVerificationMessageTemplate,
+	}
+}
+
+func (s *TestVerificationServiceSuite) TestVerify() {
+	defer gock.Off()
+	gock.New("https://api.twilio.com").
+		Reply(http.StatusNoContent).
+		BodyString("")
```

Comment:
Let's check that the mock service got the expected api call: correct phone number and the verification code (the code is available in the annotation).

Reply:
This was not easy, but I worked out a way to do it :)